client/src/components/menubar/index.tsx

```diff
@@ -72,58 +84,54 @@ interface DispatchProps {
   dispatch: AppDispatch;
 }
 export type MenuBarProps = StateProps & DispatchProps;
-interface State {
-  pendingClipPercentiles: {
-    clipPercentileMin: number;
-    clipPercentileMax: number;
-  };
-}
-class MenuBar extends React.PureComponent<MenuBarProps, State> {
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  static isValidDigitKeyEvent(e: any) {
-    /*
+
+const isValidDigitKeyEvent = (e: KeyboardEvent) => {
+  /*
     Return true if this event is necessary to enter a percent number input.
     Return false if not.
 
     Returns true for events with keys: backspace, control, alt, meta, [0-9],
     or events that don't have a key.
     */
-    if (e.key === null) return true;
-    if (e.ctrlKey || e.altKey || e.metaKey) return true;
-
-    // concept borrowed from blueprint's numericInputUtils:
-    // keys that print a single character when pressed have a `key` name of
-    // length 1. every other key has a longer `key` name (e.g. "Backspace",
-    // "ArrowUp", "Shift"). since none of those keys can print a character
-    // to the field--and since they may have important native behaviors
-    // beyond printing a character--we don't want to disable their effects.
-    const isSingleCharKey = e.key.length === 1;
-    if (!isSingleCharKey) return true;
-
-    const key = e.key.charCodeAt(0) - 48; /* "0" */
-    return key >= 0 && key <= 9;
-  }
-
-  constructor(props: MenuBarProps) {
-    super(props);
-
-    this.state = {
-      pendingClipPercentiles: INITIAL_PERCENTILES,
-    };
-  }
-
-  isClipDisabled = () => {
+  if (e.key === null) return true;
+  if (e.ctrlKey || e.altKey || e.metaKey) return true;
+
+  // concept borrowed from blueprint's numericInputUtils:
+  // keys that print a single character when pressed have a `key` name of
+  // length 1. every other key has a longer `key` name (e.g. "Backspace",
+  // "ArrowUp", "Shift"). since none of those keys can print a character
+  // to the field--and since they may have important native behaviors
+  // beyond printing a character--we don't want to disable their effects.
+  const isSingleCharKey = e.key.length === 1;
+  if (!isSingleCharKey) return true;
+
+  const key = e.key.charCodeAt(0) - 48; /* "0" */
+  return key >= 0 && key <= 9;
+};
+
+const MenuBar = ({
+  dispatch,
+  disableDiffexp,
+  clipPercentileMin: currentClipMin,
+  clipPercentileMax: currentClipMax,
+  graphInteractionMode,
+  showCentroidLabels,
+  categoricalSelection,
+  colorAccessor,
+  subsetPossible,
+  subsetResetPossible,
+  screenCap,
+}: MenuBarProps) => {
+  const [pendingClipPercentiles, setPendingClipPercentiles] =
+    useState(INITIAL_PERCENTILES);
+  const [menuBarWidthMatches, setMenuBarWidthMatches] = useState(false);
```

Comment:
discussion: I'm wondering if naming it `isVertical` would make it more consistent with where it's used (` vertical={menuBarWidthMatches}`) and declarative?